src/test/scala/com/advancedtelematic/campaigner/http/CampaignResourceSpec.scala

```diff
@@ -472,4 +485,25 @@ class CampaignResourceSpec
     val elemsWithGroupNumAssigned = seq.groupBy(_ => Gen.choose(0, n - 1).generate)
     0.until(n).map(groupNum => elemsWithGroupNumAssigned.getOrElse(groupNum, Seq.empty))
   }
+
+  private def assertRetryCampaign(retryCampaign: GetCampaign, failureCode: String, updateId: UpdateId, mainCampaignId: CampaignId): Assertion = {
+    eventually {
```

Comment:
What was the reason to wrap it in `eventually`? Shouldn't the retry campaign be available right after we get 201 from the `retry-failed` endpoint?

Reply:
The thing is that sometimes the test checks too soon and the status of the retry-campaign is `prepared` instead of `launched`, making the test flaky. My two ideas would be to wrap it in an `eventually` or to leave the campaign status out of the comparison. It just seemed easier to go with `eventually`.

Reply:
Ah, I see, good point